pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -634,154 +635,159 @@ def getData(self):
-                if view is None or not view.autoRangeEnabled()[0]:
-                    # this option presumes that x-values are in increasing order
-                    range = self.viewRect()
-                    if range is not None and len(x) > 1:
-                        # clip to visible region extended by downsampling value, assuming
-                        # uniform spacing of x-values, has O(1) performance
-                        dx = float(x[-1]-x[0]) / (len(x)-1)
-                        # workaround for slowdown from numpy deprecation issues in 1.17 to 1.20+
-                        # x0 = np.clip(int((range.left()-x[0])/dx) - 1*ds, 0, len(x)-1)
-                        # x1 = np.clip(int((range.right()-x[0])/dx) + 2*ds, 0, len(x)-1)
-                        x0 = fn.clip_scalar(int((range.left()-x[0])/dx) - 1*ds, 0, len(x)-1)
-                        x1 = fn.clip_scalar(int((range.right()-x[0])/dx) + 2*ds, 0, len(x)-1)
-
-                        # if data has been clipped too strongly (in case of non-uniform
-                        # spacing of x-values), refine the clipping region as required
-                        # worst case performance: O(log(n))
-                        # best case performance: O(1)
-                        if x[x0] > range.left():
-                            x0 = np.searchsorted(x, range.left()) - 1*ds
-                            x0 = fn.clip_scalar(x0, 0, len(x)) # workaround
-                            # x0 = np.clip(x0, 0, len(x))
-                        if x[x1] < range.right():
-                            x1 = np.searchsorted(x, range.right()) + 2*ds
-                            x1 = fn.clip_scalar(x1, 0, len(x))
-                            # x1 = np.clip(x1, 0, len(x))
-                        x = x[x0:x1]
-                        y = y[x0:x1]
-
-            if ds > 1:
-                if self.opts['downsampleMethod'] == 'subsample':
-                    x = x[::ds]
-                    y = y[::ds]
-                elif self.opts['downsampleMethod'] == 'mean':
-                    n = len(x) // ds
-                    x = x[:n*ds:ds]
-                    y = y[:n*ds].reshape(n,ds).mean(axis=1)
-                elif self.opts['downsampleMethod'] == 'peak':
-                    n = len(x) // ds
-                    x1 = np.empty((n,2))
-                    x1[:] = x[:n*ds:ds,np.newaxis]
-                    x = x1.reshape(n*2)
-                    y1 = np.empty((n,2))
-                    y2 = y[:n*ds].reshape((n, ds))
-                    y1[:,0] = y2.max(axis=1)
-                    y1[:,1] = y2.min(axis=1)
-                    y = y1.reshape(n*2)
+        if( self.xDisp is not None
+            and not (self._xViewRangeWasChanged and self.opts['clipToView'])
+            and not (self._xViewRangeWasChanged and self.opts['autoDownsample'])
+            and not (self._yViewRangeWasChanged and self.opts['dynamicRangeLimit'] is not None)
+        ):
+            return self.xDisp, self.yDisp
```

Comment:
Early return here avoids having all the following code in an indented block

Reply:
this is sort of nit-picky, would you mind moving the `and`'s from the start of the new-line to the end of the previous line?  I find that a bit easier to read myself.

```python
        if( self.xDisp is not None and
            not (self._xViewRangeWasChanged and self.opts['clipToView']) and
            not (self._xViewRangeWasChanged and self.opts['autoDownsample']) and
            not (self._yViewRangeWasChanged and self.opts['dynamicRangeLimit'] is not None)
        ):
```

Reply:
I mildly prefer my logic to be visible at the start of the line, but no strong opinion on that.
--> changed